### ISSUE TITLE
[REVIEW] R1 Review

### ISSUE THREAD

**Issue by cy99vis** [journalovi/2024-luong-graphical-models#11]
### Conflicts of interest

- [x] I declare that I have no known conflicts of interest with the authors.

### Reviewed version

2a65b3c

### Review

This paper presents an interactive account of how three different probabilistic graphical models (HMMs, MEMMs, and CRFs) work, particularly in the context of performing named entity recognition (NER) tasks. 
Overall, I think the writing was clear, and the explanations provided were helpful especially given that I am not a modelling expert.
I appreciate the discussion of each model’s limitations, and I really like the interactive boxes where the user can see the model predictions/calculations for different examples throughout the article. 
The state transition box under the MEMM section and the Calculating p(A,B,C,D,E,F) box under the CRF section, were particularly helpful for my understanding.
In general, I think the interactive components were effectively incorporated, and this article would be a great contribution to JOVI.

### Openness/Transparency

I am not sure whether the actual code the authors used to train each model and produce the results shown in the article is included in the repository. 
However, I feel the authors described the setup of each model clearly and the dataset using for training is also available online, so it seems fairly feasible to replicate and extend their work. 
The interactive article code itself is also well documented and seems easy to build on top of.

### Submission categories

- [ ] Registered Report
- [ ] Replication Study
- [x] Empirical Research - Quantitative
- [ ] Empirical Research - Qualitative
- [ ] Systems or design research
- [x] Commentary
- [ ] Systematic Literature Review

### Suggested outcome

Minor revisions: this paper requires some smaller changes, after which I am confident I would be able to endorse it.

### Requested changes

Overall, I really enjoyed this article and believe it should be accepted with minor revisions. Most of my suggestions below are minor as well:

**General**
* Perhaps throughout the article, something like a popup could be added to key terms/phrases to remind readers (e.g., in the first interactive box, “named entity” is underlined, so this could be one place for a popup reminder of the definition).
* I realized in all the static modelling diagrams, the hidden states are at the top and the observed states are at the bottom, however in the interactive examples, they are flipped. Perhaps this is something to consider making more consistent for clarity?
* I would suggest adding a brief conclusion section to summarize the key takeaways and perhaps add an interactive widget for users to explore the results of all three models side by side (similar to the precision/recall tables above or maybe tying back to the box at the top where you can see how the NER predictions differ based on the model).

**Abstract**
* In the last paragraph, I would expand briefly on this: “In this article, we’ll explore three sequence models that are remarkably successful at NER.” What is the goal in exploring these three models? To find similarities/differences in how they perform NER? Analyze performance differences? Etc.
* I would also be curious to hear more about why these three models in particular were chosen.

**I. Probabilistic Graphical Models**
* Perhaps a quick definition of what a “factor” means at the beginning of this section could be helpful to readers who are not as familiar with graphical modeling.
* A cool addition to make the paper more interactive might be to link the factorization equation and the DAG so that when a reader hovers on a node/link in the DAG for example, the corresponding components in the equation get highlighted and vice versa.
* The “Certain sampling parameters have changed” message could probably be edited slightly to be more noticeable given the plots in the background (e.g., larger/different color font? Put a colored background overlay underneath?).

**II. Hidden Markov Models**
* Very minor, but potentially, the hidden/observed layer links could be indicated by a different color in the diagrams for clarity (however, I also do like the current aesthetic of the article as well).
* I know you had the legend above with what the different tags mean (e.g., O, LOC, etc.), but perhaps it would be helpful to have this legend in a floating box throughout the relevant sections so readers don’t have to scroll back up. Additionally, I wasn’t quite sure what the B and I represented until I read more in the article.
* Small suggestion, but it may be helpful to label the left and right sides of the “Transition paths between hidden states” diagram (e.g., xi-1 on the left and xi on the right).
* I might also make a note about what the dashed lines mean in the “Emission paths from hidden states to observations” diagram.
* Perhaps under the “training” section, you could add a link in case readers want more information about how the probabilities are computed (e.g., even the same textbook that was mentioned above).
* Under the “inference” section, a quick reminder that the 81 comes from 9^2 might be helpful.
* Under “results,” maybe be interesting to include a couple examples of what an OOV token for this training set is. 
  * Also, some clarification on what “word identity” means would be helpful as well.
  * Do the empty squares in the entity length vs. OOV rate plot mean there was no data available for that specific combination? I might add a quick footnote for clarification.

**III. Maximum Entropy Markov Models***
* I wonder if there is a way to clarify the differences between HMM and MEMM more via the diagrams. For example, in the “Stately, plump Buck Mulligan came” example, the diagram itself does not look too different from the HMM example, besides the direction of the arrow. Maybe some kind of animation or something highlighting the direct modelling of the conditional probabilities for MEMMs would help clarify this point more?
* I’m also curious if there would be a way to add some more examples of what different word shapes mean in this section (maybe via another interactive or floating box)? It was helpful to see examples like U.S. / U.N. --> “X.X.” in the text, but that made me think it would be great to have a table with more examples as I was a bit confused at first not being too familiar with the concept of “shape” in this context.
* For the “results,” I personally feel it may be more helpful for comparison to put the precision of MEMM and HMM next to each other, etc., rather than having to toggle to see the results of MEMM vs. HMM.
* Is there a concrete example that could be used to explain the label bias problem? (I think I understand the idea but having an actual example with words could be helpful here.)

**IV. Conditional Random Fields**
* Minor: In the third paragraph beginning with “Acyclic…” should the “Graphs” also be capitalized? 
* Referring back to my earlier comment about the use of the word “factors,” I like that there is an explanation here, however if the term did not have the same meaning above (which I feel may be the case), it would definitely be good to clarify.
* Potentially a brief explanation or popup for what a “clique” means would be helpful as well.
* I wonder if “addressing the label bias problem” or another variation might be a more suitable header (or else it may seem like CRFs still have the same label bias issue as MEMMs on a first glance).
* It seems this section is missing an explicit “results” subheading. Was this intentional?
* Similar comments about having more side by side comparisons across the different models here.
